### ISSUE TITLE
cudnn < 6 needs to disable batchnorm codepath

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#1188]
As Arthur Szlam reports, fb-internal cudnn is still lagging behind, and giving batch-size > 1024 with batchnorm is raising an error from cudnn. Need to check for compile-time version and disable this codepath

**Comment by ngimel** [comment]
The 1024 limitation was removed in 5.1.10
